core/src/main/kotlin/org/coepi/api/v4/http/HttpResponse.kt

```diff
@@ -0,0 +1,17 @@
+package org.coepi.api.v4.http
+
+import java.nio.ByteBuffer
+
+sealed class HttpResponse(val status: Int, val body: ByteBuffer?)
```

Comment:
Does the ktor response expect a bytebuffer body or does it use string ? If latter, then I recommend keeping it as a string to prevent back-and-forth conversions. Also, we should probably add response headers since that may be required for CDN caching or you can also just use https://kotlinlang.org/api/latest/jvm/stdlib/org.w3c.fetch/-response/

Reply:
I don't think we'd want to use that type. Looks like a wrapper around a JavaScript type, including returning the Promise type from JS.

Ktor can handle all sorts of things, including text and bytes. In the byte case it actually wants a ByteArray. ByteBuffer is nice because it has a sane equals implementation, so when you use it in a data class `==` still works, and converting between ByteArray and ByteBuffer is pretty trivial. (ByteBuffer -> ByteArray isn't even a conversion; it's just returning the underlying array if one exists.)

Besides raw character and byte strings, Ktor also supports content negotiation. We could just accept/return objects and let Ktor manage deser/ser. This is actually where this whole HttpResponse type starts to make me itchy. If we were just a Ktor service I'd use the built-in content negotiation so that we don't have to do the serialization ourselves, but supporting Lambda throws a wrench in that. It's fine for now because we can safely assume the bytes in the response are JSON, but if we're going to support multiple return content types things are going to get ugly quick.

Reply:
Ok, let's see what the end state looks like after the ktor implementation.

Reply:
Oh yeah, I didn't even address cache control headers. Ktor has [some support for that](https://ktor.io/servers/features/caching-headers.html). I'd have to check what we need to do for Lambda.